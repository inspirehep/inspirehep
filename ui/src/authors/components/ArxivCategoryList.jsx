import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { List } from 'immutable';

import UnclickableTag from '../../common/components/UnclickableTag';
import InlineList from '../../common/components/InlineList';

class ArxivCategoryList extends Component {
  static renderArxivCategory(category) {
    return <UnclickableTag>{category}</UnclickableTag>;
  }

  render() {
    const { arxivCategories } = this.props;
    return (
      <InlineList
        separateItems={false}
        items={arxivCategories}
        renderItem={ArxivCategoryList.renderArxivCategory}
      />
    );
  }
}

ArxivCategoryList.propTypes = {
  arxivCategories: PropTypes.instanceOf(List),
};

ArxivCategoryList.defaultProps = {
  arxivCategories: null,
};

export default ArxivCategoryList;
