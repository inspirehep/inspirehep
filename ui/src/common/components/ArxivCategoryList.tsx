import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { List } from 'immutable';

import UnclickableTag from './UnclickableTag';
import InlineList from './InlineList';

class ArxivCategoryList extends Component {
  static renderArxivCategory(category) {
    return <UnclickableTag>{category}</UnclickableTag>;
  }

  render() {
    const { arxivCategories, wrapperClassName } = this.props;
    return (
      <InlineList
        separateItems={false}
        items={arxivCategories}
        renderItem={ArxivCategoryList.renderArxivCategory}
        wrapperClassName={wrapperClassName}
      />
    );
  }
}

ArxivCategoryList.propTypes = {
  arxivCategories: PropTypes.instanceOf(List),
  wrapperClassName: PropTypes.string,
};

ArxivCategoryList.defaultProps = {
  arxivCategories: null,
  wrapperClassName: null,
};

export default ArxivCategoryList;
