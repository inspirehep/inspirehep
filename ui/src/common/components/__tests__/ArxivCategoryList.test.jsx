import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import ArxivCategoryList from '../ArxivCategoryList';

describe('ArxivCategoryList', () => {
  it('renders arxiv categories', () => {
    const arxivCategories = fromJS(['hep-ex', 'hep-ph']);
    const wrapper = shallow(
      <ArxivCategoryList arxivCategories={arxivCategories} />
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
