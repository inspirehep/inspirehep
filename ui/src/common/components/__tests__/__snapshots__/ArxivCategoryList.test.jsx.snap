// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArxivCategoryList renders arxiv categories 1`] = `
<div
  className="__InlineList__"
>
  <ul
    className=""
  >
    <li
      key="hep-ex"
    >
      <UnclickableTag>
        hep-ex
      </UnclickableTag>
    </li>
    <li
      key="hep-ph"
    >
      <UnclickableTag>
        hep-ph
      </UnclickableTag>
    </li>
  </ul>
</div>
`;
